File: src/main/java/com/sangamesh/Fitsphere/service/FavoriteService.java
package com.sangamesh.Fitsphere.service;

import com.sangamesh.Fitsphere.dto.exercise.ExerciseSummaryDto;
import com.sangamesh.Fitsphere.dto.nutrition.FoodSummaryDto;
import com.sangamesh.Fitsphere.entity.Favorite;

import java.util.List;

public interface FavoriteService {

    void addFavorite(Long exerciseId);

    void removeFavorite(Long exerciseId);

    List<ExerciseSummaryDto> getUserFavorites();

    boolean isFavorite(Long exerciseId);

    //food favorite services
    void addFoodFavorite(Long fdcId);

    void removeFoodFavorite(Long fdcId);

    List<FoodSummaryDto> getUserFoodFavorites();

    boolean isFoodFavorite(Long fdcId);
}

File: frontend/src/services/favoriteService.js
import api from "./api";

const favoriteService = {

    // ---------- Exercises ----------

    getUserFavorites: async () => {
        const { data } = await api.get("/favorites");
        return data;
    },

    addFavorite: async (exerciseId) => {
        const { data } = await api.post(
            `/favorites/${exerciseId}`
        );
        return data;
    },

    removeFavorite: async (exerciseId) => {
        const { data } = await api.delete(
            `/favorites/${exerciseId}`
        );
        return data;
    },

    isFavorite: async (exerciseId) => {
        const { data } = await api.get(
            `/favorites/${exerciseId}/status`
        );
        return data;
    },

    // ---------- Foods ----------

    getUserFoodFavorites: async () => {
        const { data } = await api.get(
            "/favorites/foods"
        );
        return data;
    },

    addFoodFavorite: async (fdcId) => {
        const { data } = await api.post(
            `/favorites/foods/${fdcId}`
        );
        return data;
    },

    removeFoodFavorite: async (fdcId) => {
        const { data } = await api.delete(
            `/favorites/foods/${fdcId}`
        );
        return data;
    },

    isFoodFavorite: async (fdcId) => {
        const { data } = await api.get(
            `/favorites/foods/${fdcId}/status`
        );
        return data;
    }

};

export default favoriteService;